### PR TITLE
Fixes crash on accounts without overclocks unlocked

### PR DESCRIPTION
**Pull request by simon-wg**
Quick and easy solution, if no overclocks are found we simply mark all as unaquired and skip the entire process of handling them.

Note that I have not tried what happens if you add overclocks to an account without them unlocked.

### CODE REVIEW

**Comment by AnthonyMichaelTDM** [comment]
I think I'd prefer if the whole Overclocks section of the UI was greyed out, perhaps with a message saying that "Weapon OCs not unlocked yet" or something

**Comment by AnthonyMichaelTDM** [comment]
thanks for contributing!

**Comment by AnthonyMichaelTDM** [comment]
this works as the backend handling of the issue though

**Comment by simon-wg** [comment]
Good idea, I'm taking a look at that right now

**Review by AnthonyMichaelTDM — DISMISSED** [review]
LGTM

**Comment by AnthonyMichaelTDM** [comment]
could you write a test for these changes before I merge or is that not worth the effort?

**Comment by simon-wg** [comment]
wait a moment, I'm gonna add a text that tells the user that they haven't unlocked overclocks yet. That way we can avoid some unneccessary bug reports.

**Comment by AnthonyMichaelTDM** [comment]
sounds good, let me know (ping me) when it's all ready

File: src/main/python/core/controller.py [hunk 1]
```diff
@@ -624,8 +624,13 @@ def build_oc_tree(self, tree: QTreeWidgetItem) -> None:
...
         overclock_tree = self.widget.overclock_tree.invisibleRootItem()
-        self.build_oc_tree(overclock_tree)
-        self.widget.overclock_tree.sortItems(0, Qt.SortOrder.AscendingOrder)
+        if self.state_manager.get_max_promo() > 0:
+            self.build_oc_tree(overclock_tree)
+            self.widget.overclock_tree.sortItems(0, Qt.SortOrder.AscendingOrder)
+        else:
+            error_text = QTreeWidgetItem(None)
+            error_text.setText(0, "No dwarf promoted yet")
+            overclock_tree.addChild(error_text)
```
Comment (AnthonyMichaelTDM) [inline src/main/python/core/controller.py:633]: wouldn't
```suggestion
            error_text = QTreeWidgetItem(overclock_tree)
            error_text.setText(0, "No dwarf promoted yet")
```
have the same effect?

Comment (simon-wg) [reply src/main/python/core/controller.py:633]: Indeed, fixing right now

**Comment by simon-wg** [comment]
alright, that will have to do for me for today, got work in 7 hours, so can't do the testing today.

**Comment by simon-wg** [comment]
I did try the testing a little bit, but got some issues related to having to create a widget in order to test that the QTreeWidget is disabled. 
If you or @wahlp want to have a go at it feel free, but I don't think it's too overly necessary to add a test for it.

**Comment by AnthonyMichaelTDM** [comment]
That's more than fair, I'll take a look. Thanks for all the work

**Review by AnthonyMichaelTDM — DISMISSED** [review]

**Comment by AnthonyMichaelTDM** [comment]
idk what's going on here, I'm not getting this error on my system ...

**Comment by wahlp** [comment]
the third test test_issue97 is hanging on my machine too.

tried copying this from main and put it at the top and they passed.

```py
from PySide6.QtCore import QCoreApplication, Qt
QCoreApplication.setAttribute(Qt.ApplicationAttribute.AA_ShareOpenGLContexts, True)
```

**Comment by wahlp** [comment]
https://stackoverflow.com/questions/75497408/github-action-pytest-exit-code-134
can't believe this worked, nice

**Comment by AnthonyMichaelTDM** [comment]
huh, funky

**Comment by AnthonyMichaelTDM** [comment]
thanks for figuring that out though

**Review by AnthonyMichaelTDM — APPROVED** [review]

**Pull request merged**